tools/analysis/runtime.py

```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/env python
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (C) 2017, ARM Limited, Google, and contributors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import trappy
+import numpy as np
+import argparse
+
+class RunData:
+    def __init__(self, pid, comm, time):
+        self.pid = pid
+        self.comm = comm
+        self.last_start_time = time
+        self.total_time = np.float64(0.0)
+        self.start_time = -1
+        self.end_time = -1
+        self.running = 1
+        self.maxrun = -1
+
+parser = argparse.ArgumentParser(description='Analyze runtimes')
+
+parser.add_argument('--trace', dest='trace_file', action='store', required=True,
+                    help='trace file')
+
+parser.add_argument('--normalize', dest='normalize', action='store_true', default=False,
+                    help='normalize time')
+
+parser.add_argument('--rows', dest='nrows', action='store', default=20, type=int,
+                    help='normalize time')
+
+parser.add_argument('--start-time', dest='start_time', action='store', default=0, type=float,
+                    help='trace window start time')
+
+parser.add_argument('--end-time', dest='end_time', action='store', default=None, type=float,
+                    help='trace window end time')
+
+args = parser.parse_args()
+
+path_to_html = args.trace_file
+nrows = args.nrows
+
+# Hash table for runtimes
+runpids = {}
+
+starttime = None
+endtime = None
+
+def switch_cb(data):
+    global starttime, endtime
```

Comment:
If some global state is needed (more than just the latpid), it may be worth putting these callbacks into a class and keep track of the state in the instance.